### DISCUSSION TITLE
Download stop at 9%

### DISCUSSION THREAD

**Discussion by MightyMaestro** [aclap-dev/video-downloadhelper#36] [Help and Q&A]
### Your question

Ninety percent of my downloads stop at 9% downloaded. I might get the file file done, but the rest will stall at 9% downloaded. It also does not matter when I start the next download, or if there is only one or a few downloads.

![dwhelper](https://github.com/aclap-dev/video-downloadhelper/assets/97915407/57e3685e-6334-47c8-89e3-7d3d63d898a4)

### Addon & CoApp version and details

Chrome
Version 8.1.6.0
VdhCoApp: 2.0.10

**Comment by paulrouget** [comment]
I know what's going on. That 9% thing means the download is stalling.

I'm looking at it.

**Reply by MightyMaestro** [reply]
Ok. Thank you. That would be good.

**Comment by MWMarshall** [comment]
I was having the exact same problem on Mac Firefox (Version 8.1.4.0 and CoApp 2.0.10), and as a short-term band-aid installing the beta version of 8.2 improved the situation greatly.  It still stalls at a random early point maybe 10-20% of the time, but it's at least usable, and seems to do a better job of succeeding on restart on problem files.

**Reply by GrampaWildWilly** [reply]
Let me get some clarification.  Are you simply patient & wait 3, 4, 5 minutes or longer, & VDH eventually completes the download without any intervention on your part?

**Reply by MWMarshall** [reply]
No; although I haven't tested extensively with the current beta (8.2.0.17a1), as far as I can tell once a download has stalled for longer than a relatively short period (maybe 1-2 minutes), it will not progress on its own no matter how long it's left running.   I just left a stuck download up for 20 minutes and counting, and nothing has changed.  I do sometimes see a file stop download activity for a short while then start up again, but only if activity restarts within a couple of minutes, give or take; after that the hang seems to be permanent.

If I manually abort the download, delete the partial file, reload the page, and try the same download again, it will usually succeed at that point (at least, it appears to have the same ~80% chance of succeeding as a "new" download).

If this is of any interest, these stuck downloads also don't appear to time out, VDH will just sit there with the estimated time of completion gradually increasing indefinitely. The download speed displayed doesn't go to 0kB/s, it just sticks at the value at the point it stalled, which can be in the MB/s range.  I've confirmed that nothing is being written to disk (the file size and timestamp of the partial file don't change). Other downloads started while one is hung are not impacted--they will generally complete normally.

**Reply by GrampaWildWilly** [reply]
Well, the current VDH is 8.2.0.18a1 beta & the current CoApp is 2.0.15.  I suppose it wouldn't hurt to try those.  But it also wouldn't hurt for you to post at least one URL on which you have consistently had this permanent hang.  I've just been experimenting with some content in this discussion https://github.com/aclap-dev/video-downloadhelper/discussions/31 & those downloads hang for 3-4 minutes, then resume & complete successfully.  That's why I asked my question.  But if you're hanging indefinitely, that sounds like a different issue.

**Reply by MWMarshall** [reply]
That's strange; I installed the latest beta version from the VHD website earlier today, and I just now went to what I thought was the correct [beta install page](https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions/) here at Github, and I got 8.2.0.17a1 from both sources, showing a build date of 1/30. Is there a different way I should be installing the beta to get a newer version?  Or maybe Firefox has a different version number than Chrome/Edge (the release FF plugin is 8.1.4.0 rather than 8.1.6.0 you get for Edge and Chrome).

I realize now that the CoApp doesn't update on its own, so I manually installed 2.0.15.

In any case, I'm _also_ seeing what may be the same temporary stall in that other discussion (although in my case I'd guess it's more like 2 minutes than 3-4), but the perpetual hang issue does appear to be something unrelated.

I've only tried a few downloads so it's possible it's some unrelated network or site thing, but interestingly after updating the CoApp I seem to be getting different behavior now--I haven't gotten any hangs so far, but instead about 20% of download attempts are failing in a weird way--the download appears to complete, and the resulting file does play, but the file generated is quite small and has a random short segment from somewhere in the video.  As before, trying again does seem to mostly succeed.

Assuming this is a change in the response to whatever is causing these random issues, it's good that it doesn't get stuck, but it's bad that if you don't check the file you won't know that it's not actually the whole thing.

**Reply by GrampaWildWilly** [reply]
Oh.  Right.  Chrome.  Sorry.  I guess the version numbers are different.  It doesn't actually say on the page what's most recent for each browser.  In case it isn't obvious, I don't work for VDH.  I'm just another user.

And right, the CoApp doesn't auto-update.  I think the idea is that the browser extension will detect when your CoApp is too far backlevel & prompt you, but you weren't that far back so you didn't get the prompt.

Fails with a short result.  That's interesting, too.  If you read all of that other thread (it's not short) you'll see where there were some issues of corrupt packets.  If you give me a couple of URLs that are causing you trouble, I could see what happens with Firefox, and if necessary (and possible), ffmpeg.

**Reply by MightyMaestro** [reply]
No. I have just tried a download this afternoon. It has been sitting at 9% for four hours

**Reply by GrampaWildWilly** [reply]
Is that a URL behind a paywall or could I get access to it without a login?  It might be helpful if you post that URL.
